script/custom-eslint/enzyme-always-unmount/index.js

```diff
@@ -0,0 +1,27 @@
+module.exports.rules = {
+  'no-matching-unmount': context => {
+    let shallowMountNode = null;
+    let unmountNode = null;
+    return {
+      "CallExpression[callee.name='it'] > ArrowFunctionExpression Identifier[name=/shallow|mount|unmount/]": node => {
```

Comment:
Hey, cool! This doesn't match `foo.it()`.

Only improvement I can think of is to use `:function` instead of `ArrowFunctionExpression` since that'll also match `function foo() { ... }`. Not that it really matters _too_ much since we have another linting rule that calls `function` definitions an error. :man_shrugging: 

What this _won't_ do is catch it if the callback is defined outside of the `it()` call. (`it('should do a thing', testFunction)`, for example.) We can probably catch that in a later iteration if we care, though.